src/ProductConstructionService/ProductConstructionService.Api/Queue/JobsProcessor.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.Extensions.Options;
+using ProductConstructionService.Api.Queue.Jobs;
+
+namespace ProductConstructionService.Api.Queue;
+
+public class JobsProcessor(
+    ILogger<JobsProcessor> logger,
+    IOptions<JobProcessorOptions> options,
+    JobsProcessorScopeManager scopeManager,
+    QueueServiceClient queueServiceClient)
+    : BackgroundService
+{
+    private readonly ILogger<JobsProcessor> _logger = logger;
+    private readonly IOptions<JobProcessorOptions> _options = options;
+    private readonly JobsProcessorScopeManager _scopeManager = scopeManager;
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        await ProcessJobsAsync(cancellationToken);
+    }
+
+    private async Task ProcessJobsAsync(CancellationToken cancellationToken)
+    {
+        QueueClient queueClient = queueServiceClient.GetQueueClient(_options.Value.JobQueueName);
+        _logger.LogInformation("Starting to process PCS jobs {queueName}", _options.Value.JobQueueName);
+        while (!cancellationToken.IsCancellationRequested)
+        {
+            using (_scopeManager.BeginJobScopeWhenReady(cancellationToken))
+            {
+                try
+                {
+                    await ReadAndProcessJobAsync(queueClient, cancellationToken);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "An unexpected exception occurred during Pcs job processing");
+                }
+            }
+        }   
+    }
+
+    private async Task ReadAndProcessJobAsync(QueueClient queueClient, CancellationToken cancellationToken)
+    {
+        QueueMessage message = await queueClient.ReceiveMessageAsync(_options.Value.QueueMessageInvisibilityTime, cancellationToken);
+
+        if (message?.Body == null)
+        {
+            // Queue is empty, wait a bit
+            _logger.LogDebug("Queue {queueName} is empty. Sleeping for {sleepingTime} seconds", _options.Value.JobQueueName, (int)_options.Value.QueuePollTimeout.TotalSeconds);
+            await Task.Delay(_options.Value.QueuePollTimeout, cancellationToken);
+            return;
+        }
+
+        var job = message.Body.ToObjectFromJson<Job>();
+
+        try
+        {
+            ProcessJob(job);
```

Comment:
Log that we're starting an attempt X of a job with ID Y, possibly also log the type of the job

Reply:
added